templates/digital-asset-template/move/sources/launchpad.move

```diff
@@ -262,12 +264,30 @@ module launchpad_addr::launchpad {
         let stage = &mint_stage::execute_earliest_stage(sender, collection_obj, 1);
         assert!(option::is_some(stage), E_NO_ACTIVE_STAGES);
 
-        let mint_fee = get_mint_fee(collection_obj, *option::borrow(stage));
+        let mint_fee = get_mint_fee_per_nft(collection_obj, *option::borrow(stage));
         pay_for_mint(sender, mint_fee);
 
         mint_nft_internal(sender_addr, collection_obj, mint_fee);
     }
 
+    public entry fun batch_mint_nft(
+        sender: &signer,
+        collection_obj: Object<Collection>,
+        amount: u64,
+    ) acquires CollectionConfig, CollectionOwnerObjConfig, Config {
+        let sender_addr = signer::address_of(sender);
+
+        let stage = &mint_stage::execute_earliest_stage(sender, collection_obj, amount);
+        assert!(option::is_some(stage), E_NO_ACTIVE_STAGES);
+
+        let mint_fee_per_nft = get_mint_fee_per_nft(collection_obj, *option::borrow(stage)) * amount;
+        pay_for_mint(sender, mint_fee_per_nft * amount);
```

Comment:
do we need to check if the signer has enough balance to pay for each nft?

Reply:
it will throw not enough balance error if so